### PR TITLE
Add Polyglot Support in Runtime Tests

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

`runtime-with-instruments` project sets `-Dgraalvm.locatorDisabled=true` that disables the discovery of available polyglot languages (installed with `gu`). On the other hand, enabling locator makes polyglot languages available, but also makes the program classes and the test classes loaded with different classloaders. This way we're unable to use `EnsoContext` in tests to observe internal context state (there is an exception when you try to cast to `EnsoContext`).

The solution is to move tests with enabled polyglot support, but disabled `EnsoContext` introspection to a separate project.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] ~If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.~

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1569,6 +1570,55 @@ lazy val `runtime-with-instruments` =
...
+          s"-Dtruffle.class.path.append=$appendClasspath"
+        )
+      },
+      Test / fork := true,
+      Test / envVars ++= distributionEnvironmentOverrides ++ Map(
+        "ENSO_TEST_DISABLE_IR_CACHE" -> "false"
+      ),
+      libraryDependencies ++= Seq(
+        "org.scalatest"      %% "scalatest"             % scalatestVersion % Test,
+        "org.graalvm.truffle" % "truffle-api"           % graalVersion     % "provided",
```
Comment (JaroslavTulach) [inline build.sbt]: Are these two truffle libraries needed at all? I believe they can be removed.

File: engine/polyglot-api/src/main/java/org/enso/polyglot/RuntimeOptions.java [hunk 2]
```diff
@@ -91,6 +91,13 @@ public class RuntimeOptions {
       OptionDescriptor.newBuilder(USE_GLOBAL_IR_CACHE_LOCATION_KEY, USE_GLOBAL_IR_CACHE_LOCATION)
           .build();
 
+  public static final String ENABLE_EXECUTION_TIMER = optionName("enableExecutionTimer");
+  public static final OptionKey<Boolean> ENABLE_EXECUTION_TIMER_KEY = new OptionKey<>(true);
+  private static final OptionDescriptor ENABLE_EXECUTION_TIMER_DESCRIPTOR =
+      OptionDescriptor.newBuilder(ENABLE_EXECUTION_TIMER_KEY, ENABLE_EXECUTION_TIMER)
```
Comment (JaroslavTulach) [inline engine/polyglot-api/src/main/java/org/enso/polyglot/RuntimeOptions.java]: Specify `OptionCategory.INTERNAL`

File: engine/runtime-with-polyglot/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala [hunk 3]
```diff
@@ -1,10 +1,8 @@
 package org.enso.interpreter.test.instrument
 
-import org.enso.distribution.FileSystem
+import org.apache.commons.io.FileUtils
 import org.enso.distribution.locking.ThreadSafeFileLockManager
```
Comment (JaroslavTulach) [inline engine/runtime-with-polyglot/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala:4]: This test shall verify JavaScript is enabled somehow.

File: engine/runtime-with-polyglot/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala [hunk 4]
```diff
@@ -27,20 +25,14 @@ class RuntimeVisualizationsTest
     with Matchers
```
Comment (JaroslavTulach) [inline engine/runtime-with-polyglot/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala:25]: Can we also have a `.java` test ;-? Only then `.enso-sources*` files are generated and IGV can open the project.

**Review by JaroslavTulach — APPROVED** [review]

**Comment by Akirathan** [comment]
Does this mean that we will also be able to use other truffle langs in java tests in `org.enso.interpreter.test` package?

**Comment by 4e6** [comment]
> Does this mean that we will also be able to use other truffle langs in java tests in org.enso.interpreter.test package?

yes, all the languages installed by `gu` are available

File: engine/runtime-with-polyglot/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala (hunk 3 above)
Comment (4e6) [reply engine/runtime-with-polyglot/src/test/scala/org/enso/interpreter/test/instrument/RuntimeVisualizationsTest.scala:4]: I enabled the test `run internal IDE visualisation preprocessor catching error` that runs JS

**Review by hubertp — COMMENTED** [review]
lgtm!

**Pull request merged**